be/src/olap/tablet_schema.cpp

```diff
@@ -521,7 +521,7 @@ vectorized::AggregateFunctionPtr TabletColumn::get_aggregate_function_union(
 
 vectorized::AggregateFunctionPtr TabletColumn::get_aggregate_function(std::string suffix) const {
     auto type = vectorized::DataTypeFactory::instance().create_data_type(*this);
-    if (type && type->get_type_as_primitive_type() == PrimitiveType::TYPE_AGG_STATE) {
+    if (type && type->get_type_as_type_descriptor().type == PrimitiveType::TYPE_AGG_STATE) {
```

Comment:
what's the difference?

Reply:
type_descriptor has structured info like it's sub column type, but primitive type does not